### PR TITLE
Add watch options.

### PR DESCRIPTION
**Pull request by jortel**
Add support for watch options.
- **snapshot** _(bool)_ - List models in the collection and report as `Created` events for each before reporting `Parity`.

Making the snapshot optional will eliminate thousands of unnecessary watch events for handlers that don't want the snapshot.

Snapshot (list of models) explicitly passed to Watch.Start() instead of the watch assuming the first iterator in the queue is the snapshot.

Grab the read lock on the DB at in Client.Watch().  This ensures no other client methods can queue events _before_ the snapshot.

### CODE REVIEW

File: pkg/inventory/model/client.go [hunk 1]
```diff
@@ -334,52 +336,25 @@ func (r *Client) Watch(model Model, handler EventHandler) (w *Watch, err error)
...
+		if err != nil {
 			return
 		}
+	} else {
+		snapshot = &fb.EmptyIterator{}
 	}
-	w.notify(list.Iter())
-	w.Start()
+
+	w.Start(snapshot)
```
Comment (jortel) [inline pkg/inventory/model/client.go:350]: Pass the (snapshot) list of models to the watch.
This is more explicit and better separation of concerns.  Mainly knowledge of now `Event` and model objects are written to the file-backed list.

File: pkg/inventory/model/journal.go [hunk 2]
```diff
@@ -169,15 +179,33 @@ func (w *Watch) Start() {
...
 		for itr := range w.queue {
 			for {
 				event := Event{}
 				event, hasNext, err := w.next(itr)
 				if err != nil {
-					w.log.V(3).Error(err, "next() failed.")
+					w.log.Error(err, "next() failed.")
 					w.Handler.Error(err)
-					break
+					return
```
Comment (jortel) [inline pkg/inventory/model/journal.go:208]: An (unlikely) error iterating is handled by informing the `Handler`.  An error _should_ end the watch.  Error recovery is the responsibility of the handler which will likely re-create the watch.

File: pkg/inventory/model/journal.go [hunk 3]
```diff
@@ -197,14 +225,12 @@ func (w *Watch) Start() {
 				case Deleted:
 					w.Handler.Deleted(event)
 				default:
-					w.Handler.Error(liberr.New("unknown action"))
+					w.log.Info(
+						"unknown action.",
+						"event",
+						event.String())
```
Comment (jortel) [inline pkg/inventory/model/journal.go:231]: Logged but not the concern of the `Handler`.

File: pkg/inventory/web/client.go [hunk 4]
```diff
@@ -233,8 +241,12 @@ func (r *Client) Watch(
 	if ht, cast := r.Transport.(*http.Transport); cast {
 		dialer.TLSClientConfig = ht.TLSClientConfig
 	}
+	options := []string{""}
+	if h.Options().Snapshot {
+		options = []string{WatchSnapshot}
```
Comment (jortel) [inline pkg/inventory/web/client.go:246]: Watch options are passed in the value of `X-Watch` header (values).

**Review by mansam — APPROVED** [review]
LGTM

**Pull request merged**